test/common/ops/spells.js

```diff
@@ -35,4 +35,21 @@ describe('shared.ops.spells', () => {
       done();
     }
   });
+
+  it('Issue #12361: returns an error if chilling frost has already been cast', async () => {
+    await user.update({
```

Comment:
sorry, the test structure here should follow the previously existing test in this file (the one for healer) because in integration tests we simulate actual HTTP calls to the API while here we call the cast function directly, so something like this should work

```
    user.stats.class = 'wizard';
    user.stats.lvl = 15;
    user.stats.mp = 400;
    uster.stats.buffs.streaks = true;

    const spell = spells.wizard.frost;

    try {
      spell.cast(user, null, { language: 'en' });
    } catch (err) {
      expect(err).to.be.an.instanceof(BadRequest);
      expect(err.message).to.equal(i18n.t('spellAlreadyCast'));
      expect(user.stats.mp).to.eql(400);

      done();
    }
```

You'll also need to import BadRequest alongside notauthorized at the top of this file

Reply:
also this isn't an async function so the async at the top should be removed

Reply:
Thanks! Yeah, I was able to figure that out from the message. The tests should pass now.

Reply:
@paglias A doubt. What is `done` here? I noticed that we are using the same name for the arrow function in the common test.

Reply:
I am always prone to lint errors. :laughing: :see_no_evil:

Reply:
lint errors are bit difficult to catch unless you have a plugin for your code editor that highlights them , depending on the code editor you're using there might be one available

regarding done: when Mocha (our test runner) sees a function with the `done` parameter it is there to tell it that until the done parameter is called (it's a function) the test hasn't finished running.

It's used to signal test completion for tests with async callback usually as otherwise it wouldn't be possible to know when the test has finished running. See https://mochajs.org/#asynchronous-code

In this case it's used to make sure that the `catch` block is called as `done()` is called only inside the `catch` so if no error occurs in `spell.cast()` the test would never end and Mocha would signal a problem